src/visor/db.go

```diff
@@ -29,7 +31,35 @@ type ErrCorruptDB struct {
 	error
 }
 
-// CheckDatabase checks the database for corruption
+// corrputedBlocks is used for recording corrupted blocks concurrently
+type corruptedBlocks struct {
+	v    map[uint64]struct{}
+	lock sync.Mutex
+}
+
+func newCorruptedBlocks() *corruptedBlocks {
+	return &corruptedBlocks{
+		v: make(map[uint64]struct{}),
+	}
+}
+
+func (cb *corruptedBlocks) Store(seq uint64) {
+	cb.lock.Lock()
+	cb.v[seq] = struct{}{}
+	cb.lock.Unlock()
+}
+
+func (cb *corruptedBlocks) BlockSeqs() []uint64 {
+	cb.lock.Lock()
+	var seqs []uint64
+	for seq := range cb.v {
+		seqs = append(seqs, seq)
+	}
+	cb.lock.Unlock()
+	return seqs
+}
+
+// CheckDatabase checks the database for corruption, rebuild history if corrupted
```

Comment:
Caller should choose to do history rebuild based upon corrupted state

Reply:
What is the corrupted state mean? We only rebuild when the error is `ErrHistoryDBCorrupted` type